pkgs/development/tools/hover/default.nix

```
@@ -0,0 +1,77 @@
+{ buildGoModule, fetchFromGitHub, fetchurl, glfw, libGL, libX11, libXcursor, libXi,
+  libXinerama, libXrandr, libXxf86vm, makeWrapper, pkgconfig, stdenv, unzip}:
+
+buildGoModule rec {
+  pname = "hover";
+  version = "unstable-2020-01-23";
+
+  subPackages = [ "." ];
+
+  modSha256 = "019lg3s86965jyqfy83hb52y5rb6g91prsg3ckjxbcnca68lsdja";
+
+  src = fetchFromGitHub {
+    rev = "a6f11fe3c4946c9e7cce75360902bd79a077ccf4";
+    owner = "go-flutter-desktop";
+    repo = pname;
+    sha256 = "0pwdzd34zcbyspl7qiwfcrppr6zxp2zliqg9n8s6s3l6k7mffl9j";
+  };
+
+  libFlutterName = "libflutter_engine" + stdenv.targetPlatform.extensions.sharedLibrary;
+
+  libFlutter = if (stdenv.isDarwin) then
+    fetchurl {
+      name = "libflutter_engine.zip";
+      url = "https://storage.googleapis.com/flutter_infra/flutter/051762798888ebea4666c5f9c9f4c0e4d3b3b481/darwin-x64/FlutterEmbedder.framework.zip";
+      sha256 = "1m5dq6mw51knngy080idh0394xd9h8zkbam275k0n3i2763hha2b";
+    }
+  else
+    fetchurl {
+      name = "libflutter_engine.zip";
+      url = "https://storage.googleapis.com/flutter_infra/flutter/051762798888ebea4666c5f9c9f4c0e4d3b3b481/linux-x64/linux-x64-embedder";
+      sha256 = "15p6cdvi6rajjs6vv79y7vsrwwpg2kk9qlg14891wrmdn9nvpqb4";
+    };
+
+  nativeBuildInputs = [ pkgconfig unzip makeWrapper ];
+
+  buildInputs = [
+    glfw
+    libX11
+    libXcursor
+    libXinerama
+    libXrandr
+    libXxf86vm
+    libXi
+    libGL
+  ];
+
+  patches = [
+    ./fix-assets-path.patch
+    ./fix-build-docker.patch
+  ];
+  postPatch = ''
+    sed -i 's|@assetsFolder@|'"''${bin}/share/assets"'|g' internal/fileutils/assets.go
+  '';
+
+  preBuild = ''
+     unzip $libFlutter
+     lib=$(mktemp -d)
+     mv ${libFlutterName} $lib/
+     export NIX_LDFLAGS="$NIX_LDFLAGS -L$lib"
+  '';
+
+  postInstall = ''
+    mkdir -p $out/share
+    cp -r assets $out/share/assets
+    chmod -R a+rx $out/share/assets
+
+    wrapProgram "$out/bin/hover" --prefix LD_LIBRARY_PATH : ${stdenv.lib.makeLibraryPath [ glfw libX11 libXcursor libXinerama libXrandr libXxf86vm libXi libGL]}
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A build tool to run Flutter applications on desktop";
+    homepage = "https://github.com/go-flutter-desktop/hover";
+    license = licenses.bsd3;
+    maintainers = [ maintainers.ericdallo ];
+    platforms = platforms.unix ++ platforms.darwin;
```

if someone else wants to add darwin support, then can:
```suggestion
    platforms = platforms.linux;
```

```
Archive:  /nix/store/77xjx6rilnil2c9vh8rgbard1ry924mf-libflutter_engine.zip
  inflating: FlutterEmbedder.framework.zip
mv: cannot stat 'libflutter_engine.dylib': No such file or directory
```